frontend/src/tests/ForestClientInput.spec.ts

```diff
@@ -86,20 +118,310 @@ describe('ForestClientInput', () => {
+    });
+
+    it('should raise error for duplicate user input', async () => {
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+
+        const verifyButton = wrapper.find("[aria-label='Add Client Numbers']");
+        await verifyButton.trigger('click');
+        await flushPromises();
+
+        // Check if the forest client number has been added to the card
+        expect(wrapper.html().includes('Verified Client ID information')).toBe(true);
+
+        // Try to add the same client number again
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+
+        await verifyButton.trigger('click');
+        await flushPromises();
+
+        expect(wrapper.find('#forestClientInputValidationError').element.textContent).contain(`Client ID ${TEST_SUCCESS_FOREST_CLIENT_NUMBER} has already been added.`);
+        // The input field should still display the duplicate input
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        // Expect the card to have just one entry
+        expect(wrapper.findAll('#forest-client-name')).toHaveLength(1);
+    });
+
+    it('should add multiple active forest client numbers', async () => {
+        const multipleSuccessInputs = `${TEST_SUCCESS_FOREST_CLIENT_NUMBER},${TEST_SUCCESS_FOREST_CLIENT_NUMBER_2},${TEST_SUCCESS_FOREST_CLIENT_NUMBER_3}`
+        await input.setValue(multipleSuccessInputs);
+        expect(inputField.value).toBe(multipleSuccessInputs);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        const setVerifiedForestClients = wrapper.emitted('setVerifiedForestClients');
+        // setVerifiedForestClients has been called when verification is successful
+        expect(wrapper.emitted()).toHaveProperty('setVerifiedForestClients');
+        expect(setVerifiedForestClients).toHaveLength(3);
+        // The first entry in the forest client card should be equal to the first active input
+        expect(
+            setVerifiedForestClients
+                ? setVerifiedForestClients[0][0]
+                : undefined
+        ).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        // The second entry in the forest client card should be equal to the second active input
+        expect(
+            setVerifiedForestClients
+                ? setVerifiedForestClients[1][0]
+                : undefined
+        ).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2);
+        // The third entry in the forest client card should be equal to the third active input
+        expect(
+            setVerifiedForestClients
+                ? setVerifiedForestClients[2][0]
+                : undefined
+        ).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER_3);
```

Comment:
Probably you missed my previous comment. Since you already know `setVerifiedForestClients` is not `undefined` in this case, you can do: `expect(setVerifiedForestClients![2][0]).toEqual(TEST_SUCCESS_FOREST_CLIENT_NUMBER_3);`
And, some as other part of the `expect` as well, please scan your file and adjust them.
Thanks.